test/uk/gov/hmrc/traderservices/controllers/VesselDetailsFormSpec.scala

```diff
@@ -0,0 +1,133 @@
+    }
+
+    "report an error when vesselName is invalid" in {
+      val input = formInput.updated("vesselName", "$$$")
+      form.bind(input).value shouldBe None
+      form.bind(input).errors should haveOnlyErrors(
+        FormError("vesselName", "error.vesselName.invalid-characters")
+      )
+    }
+
+    "report an error when dateOfArrival is missing" in {
+      val input =
+        formInput
+          .updated("dateOfArrival.year", "")
+          .updated("dateOfArrival.month", "")
+          .updated("dateOfArrival.day", "")
+      form.bind(input).value shouldBe None
+      form.bind(input).errors should haveOnlyErrors(FormError("dateOfArrival", "error.dateOfArrival.required"))
+    }
+
+    "report an error when dateOfArrival is partially missing" in {
+      val input = formInput
+        .updated("dateOfArrival.year", "")
+        .updated("dateOfArrival.month", "")
+      form.bind(input).value shouldBe None
+      form.bind(input).errors should haveOnlyErrors(
+        FormError("dateOfArrival", "error.dateOfArrival.required-year"),
+        FormError("dateOfArrival", "error.dateOfArrival.required-month")
+      )
+    }
+
+    "report an error when dateOfArrival is invalid" in {
+      val input = formInput
+        .updated("dateOfArrival.year", "202a")
+        .updated("dateOfArrival.month", "13")
+      form.bind(input).value shouldBe None
+      form.bind(input).errors should haveOnlyErrors(
+        FormError("dateOfArrival", "error.dateOfArrival.invalid-year-digits"),
+        FormError("dateOfArrival", "error.dateOfArrival.invalid-month-value")
+      )
+    }
+
+    "report an error when timeOfArrival is missing" in {
+      val input = formInput
+        .updated("timeOfArrival.hour", "")
+        .updated("timeOfArrival.minutes", "")
+        .updated("timeOfArrival.period", "")
+      form.bind(input).value shouldBe None
+      form.bind(input).errors should haveOnlyErrors(
+        FormError("timeOfArrival", "error.timeOfArrival.required")
+      )
+    }
+
+    "report an error when timeOfArrival is partially missing" in {
+      val input = formInput
+        .updated("timeOfArrival.hour", "")
+        .updated("timeOfArrival.minutes", "")
+      form.bind(input).value shouldBe None
+      form.bind(input).errors should haveOnlyErrors(
+        FormError("timeOfArrival", "error.timeOfArrival.required")
+      )
+    }
+
+    "report an error when timeOfArrival is invalid" in {
+      val input = formInput
+        .updated("timeOfArrival.hour", "00")
+        .updated("timeOfArrival.minutes", "60")
+        .updated("timeOfArrival.period", "ma")
+      form.bind(input).value shouldBe None
+      form.bind(input).errors should haveOnlyErrors(
+        FormError("timeOfArrival", "error.timeOfArrival.invalid-hour-value"),
+        FormError("timeOfArrival", "error.timeOfArrival.invalid-minutes-value"),
+        FormError("timeOfArrival", "error.timeOfArrival.invalid-period-value")
+      )
+    }
```

Comment:
These tests seem fine though one thing I did notice was just that the error messages for incorrect digits all say "Date" of... etc, should they maybe have their respective names ie. day/month/year. 

(messages file)
<img width="675" alt="image" src="https://user-images.githubusercontent.com/55190161/95764319-35b49c00-0ca8-11eb-85d2-42520e802297.png">


<img width="740" alt="image" src="https://user-images.githubusercontent.com/55190161/95764385-4fee7a00-0ca8-11eb-805d-32e7317fc570.png">